lil-lab/recnet
# [recnet-web][v.1.11.1] Refactor page authorization logic

## Description



- Build a shared HOC (high-order component) `withAuthRequired` that handle the authentication control and role-based access control. Pages which require authentication can directly be wrapped in that HOC so that duplicated logic won’t be written again.
The "deactivate" feature can also build in this HOC to redirect user.
[Design docs for more context](https://www.notion.so/withServerSideAuth-HOC-for-page-and-layout-317a53a14a8449608d1c1a2a9082e0bf?pvs=4)
- Move another HOC `withSuspense` to `components/hoc` folder


## Related Issue





N/A

## Notes


N/A

## Test

All behaviors should remain unchanged.

When not logged in,

- Try access above `/user/following`, `/feeds`, and `/admin` by typing in url
- Should be redirect to landing page

when logged in as non-admin

- Try access `/admin`
- Shouldn’t be able to access. Should see 404 not found instead.

## TODO

- [x] Paste the testing link: https://recnet-du8zm1780-recnet-542617e7.vercel.app/
